src/main/webapp/app/overview/exercise-details/course-exercise-details.component.html

```diff
@@ -33,18 +33,20 @@
                 <span class="text-muted" *ngSwitchCase="'quiz-active'">{{ 'artemisApp.courseOverview.exerciseList.userParticipatingShort' | translate }}</span>
                 <span class="text-muted" *ngSwitchCase="'quiz-submitted'">{{ 'artemisApp.courseOverview.exerciseList.userSubmittedShort' | translate }}</span>
             </div>
-            <span *ngSwitchCase="'initialized'">
-                <jhi-updating-result class="mr-2" [exercise]="exercise" [participation]="studentParticipation" [showGradedBadge]="true"></jhi-updating-result>
-            </span>
-            <span *ngSwitchCase="'inactive'">
-                <jhi-updating-result class="mr-2" [exercise]="exercise" [participation]="studentParticipation" [showGradedBadge]="true"></jhi-updating-result>
-            </span>
-            <span *ngSwitchCase="'quiz-finished'">
-                <jhi-updating-result class="mr-2" [exercise]="exercise" [participation]="studentParticipation" [showGradedBadge]="true"></jhi-updating-result>
-            </span>
-            <span *ngSwitchCase="'exercise-submitted'">
-                <jhi-updating-result class="mr-2" [exercise]="exercise" [participation]="studentParticipation" [showGradedBadge]="true"></jhi-updating-result>
-            </span>
+            <div class="my-2">
+                <span *ngSwitchCase="'initialized'">
```

Comment:
This one is not related to the feature, but I noticed that this switch case can be simplified to a one expression like so `*ngIf=" ['initialized', 'inactive', ...].includes(exercise.participationStatus)"`

Reply:
Oh, good catch. Weird code duplication indeed. Will refactor it.

Reply:
Done.

Reply:
Are you sure that you did not change the logic here? And did you test that exactly these cases work as before? I am not sure if you can combine `ngSwitch` and `ngIf` like that.

Reply:
I tested that for all these cases `['initialized', 'inactive', 'quiz-finished', 'exercise-submitted']` the updating result component is rendered and I picked a couple of other cases not in that list and verified that the updating result component is not rendered for them.

Combining `ngSwitch` and `ngIf` should not be an issue in this case since the `ngIf` stands on its own and does not depend on the switch context. It explicitly references `exercise.participationStatus` as compared to the implicit reference that happens in the case of the `ngSwitchCase` statements.

In fact, this construct already exists just below where `ng-container` uses `ngIf` within the `ngSwitch` scope.